shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionReducer.kt

```diff
@@ -227,8 +201,73 @@ class StepCompletionReducer(private val stepRoute: StepRoute) : StateReducer<Sta
                 )
                 state to setOf(Action.LogAnalyticEvent(event))
             }
+            is InternalMessage.FetchNextInterviewStepResultSuccess ->
+                handleFetchNextInterviewStepResultSuccess(state, message)
+            is InternalMessage.FetchNextInterviewStepResultError -> {
+                state.copy(isPracticingLoading = false) to setOf(
+                    Action.ViewAction.ShowStartPracticingError(message.errorMessage)
+                )
+            }
         } ?: (state to emptySet())
 
+    private fun handleContinuePracticingClicked(state: State) =
+        if (!state.isPracticingLoading) {
+            val analyticEvent = StepCompletionClickedContinueHyperskillAnalyticEvent(
+                route = stepRoute.analyticRoute
+            )
+            state.copy(
+                isPracticingLoading = when (state.continueButtonAction) {
+                    is ContinueButtonAction.CheckTopicCompletion,
+                    ContinueButtonAction.FetchNextInterviewStep ->
+                        true
+                    ContinueButtonAction.NavigateToBack,
+                    ContinueButtonAction.NavigateToStudyPlan ->
+                        false
+                }
+            ) to setOf(
+                Action.LogAnalyticEvent(analyticEvent),
+                when (state.continueButtonAction) {
+                    ContinueButtonAction.NavigateToBack -> Action.ViewAction.NavigateTo.Back
+                    ContinueButtonAction.NavigateToStudyPlan -> Action.ViewAction.NavigateTo.StudyPlan
+                    ContinueButtonAction.CheckTopicCompletion -> state.currentStep.topic?.let {
+                        Action.CheckTopicCompletionStatus(it)
+                    } ?: Action.ViewAction.NavigateTo.Back
+                    ContinueButtonAction.FetchNextInterviewStep -> InternalAction.FetchNextInterviewStep
+                }
+            )
+        } else {
+            null
+        }
+
+    private fun handleFetchNextInterviewStepResultSuccess(
+        state: State,
+        message: InternalMessage.FetchNextInterviewStepResultSuccess
+    ): StepCompletionReducerResult =
+        state.copy(isPracticingLoading = false) to
+            setOf(
+                if (message.newStepRoute != null) {
+                    Action.ViewAction.ReloadStep(message.newStepRoute)
+                } else {
+                    Action.ViewAction.NavigateTo.Back
```

Comment:
When `message.newStepRoute == null` than I suppose we should show a modal with success message:

> _При решении задачи и обработки этого события в StepCompletionFeature добавить проверку - если StepRoute это InterviewPreparation, то удалять последнюю задачу из InterviewStepsStateRepository и открывать последнюю после удаления. Если после удаления последней задачи в списке не осталось задач - выбрасывать ViewAction, который покажет модалку с поздравлением решения задач_

<img width="371" alt="Screenshot 2023-12-29 at 7 13 11 PM" src="https://github.com/hyperskill/mobile-app/assets/8586063/244b0e57-b0d0-4790-8cde-a3e61a2ea68c">